File: java/fizzbuzz.java
import java.util.Scanner;

public class fizzbuzz {
	static int getGCD(int a, int b) {
		return b == 0 ? a : getGCD(b, a % b);
	}

	static int getLCM(int a, int b) {
		return a * b / getGCD(a, b);
	}

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int X = scan.nextInt();
		int Y = scan.nextInt();
		int N = scan.nextInt();
		for (int i = 1; i <= N; i++) {
			if (i % getLCM(X, Y) == 0)
				System.out.println("FizzBuzz");
			else if (i % X == 0)
				System.out.println("Fizz");
			else if (i % Y == 0)
				System.out.println("Buzz");
			else
				System.out.println(i);
		}
		scan.close();
	}
}

File: ts/src/fizzbuzz.ts
import readline = require('readline')

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    let [X, Y, N] = line.split(' ').map(Number)
    X ??= 1
    Y ??= 1
    N ??= 1

    for (let i = 1; i <= N; i++) {
      if (i % X === 0 && i % Y === 0) {
        console.log('FizzBuzz')
      } else if (i % X === 0) {
        console.log('Fizz')
      } else if (i % Y === 0) {
        console.log('Buzz')
      } else {
        console.log(i)
      }
    }
  })
  .on('close', () => {
    process.exit(0)
  })
